plugins/services/src/js/components/HighOrderServiceConfiguration.js

```diff
@@ -0,0 +1,31 @@
+import React from "react";
+
+import Pod from "../structs/Pod";
+import PodConfigurationContainer
+  from "../containers/pod-configuration/PodConfigurationContainer";
+import Service from "../structs/Service";
+import ServiceConfigurationContainer
+  from "../containers/service-configuration/ServiceConfigurationContainer";
+
+const HighOrderServiceConfiguration = props => {
+  const { pod, errors, onEditClick, service } = props;
+  if (pod != null) {
+    return <PodConfigurationContainer pod={pod} />;
+  }
+
+  return (
+    <ServiceConfigurationContainer
+      errors={errors}
+      onEditClick={onEditClick}
+      service={service}
+    />
+  );
+};
+
+HighOrderServiceConfiguration.propTypes = {
```

Comment:
I would change the component interface to only accept one service property that could be either a pod or  an app or frame work. It's a good practice to reduce the number of inputs to create an easy to use component. Please note the Pod inherits all it's properties from Service.